src/btree/row_srch.c

```diff
@@ -648,14 +641,18 @@ __wt_row_search(WT_CURSOR_BTREE *cbt, WT_ITEM *srch_key, bool insert, WT_REF *le
         }
     else if (collator == NULL) {
         /*
+         * In some cases we expect we're comparing more than a few keys with matching prefixes, so
+         * it's faster to avoid the memory fetches by skipping over those prefixes. That's done by
+         * tracking the length of the prefix match for the lowest and highest keys.
+         *
          * Reset the skipped prefix counts; we'd normally expect the parent's skipped prefix values
          * to be larger than the child's values and so we'd only increase them as we walk down the
          * tree (in other words, if we can skip N bytes on the parent, we can skip at least N bytes
-         * on the child). However, leaf pages at the end of the tree can be extended, causing the
-         * parent's search to be wrong for the child. We only need to reset the high count, the page
-         * can only be extended so the low count is still correct.
+         * on the child). However, if a child internal page was split up into the parent, the child
+         * page's key space will have been truncated, and the values from the parent's search may be
```

Comment:
I think we should use ancestor instead of just parent, which is more accurate.

Reply:
Updated in [5642f0d](https://github.com/wiredtiger/wiredtiger/pull/9152/commits/5642f0d63871cc24f22d47a48b3d127e6cf0e715)